setup.mjs

```diff
@@ -535,19 +536,29 @@ function interceptOfflineProgress() {
                 return resolve();
             });
         } else {
-            return originalMethod();
+            return originalMethod(loopTime);
         }
     });
 
     // Help processOffline differentiate between time skip call and offline progress call.
-    ctx.patch(Game, "testForOffline").replace((originalMethod, hours) => {
+    ctx.patch(Game, "testForOffline").replace((originalMethod, timeToGoBack) => {
         // Everything from the original testForOffline().
         return __awaiter(game, void 0, void 0, function* () {
             game.stopMainLoop();
-            game.tickTimestamp -= hours * msPerHour;
+            game.tickTimestamp -= timeToGoBack * msPerHour;
+            game.skills.forEach((skill)=>
+                {
+                    skill.timeToLevelTracker.clear();
+                    skill.timeToLevelTicks = 0;
+                    skill.timeToLevelPercentStart = skill.nextAbyssalLevelProgress;
+                }
+            );
```

Comment:
This is new to me, thanks for adding this

Reply:
Yea this is part of the new update. We also don't process offline ticks like it was before, even though this code looks similar. Instead I am triggering the game into thinking it had started processing them and then it will take as long as it needs after this function finishes to do that. Before it would process all offline tickets before this function ended.